### PR TITLE
`@slack/types` Full JSdoc and upcoming major version breaking change TODOs as comments

### PR DESCRIPTION
**Pull request by filmaj**
###  Summary

This PR:

- adds JSdocs to all the things, mirroring as much content as possible from api.slack.com
- adds some missing properties (like `response_url_enabled` on the `ChannelsSelect` BK element)
- Adds comments of the form `TODO: breaking change` in areas where I would like to make future breaking changes in the next major version
- Specifically deprecates:
  -  the old `WorkflowStep` type for legacy Steps from Apps
  - `Confirm` in favour of a new `ConfirmationDialog` interface (for consistency of terminology)
  - `Action` in favour of a new `Actionable` interface (for consistency of terminology)

I'd love for the team to review. No need to review the JSdocs in detail, but if the team could review the various "TODO" comments I added and see if that direction makes sense, that would be appreciated.

### CODE REVIEW

File: packages/types/src/block-kit/blocks.ts [hunk 1]
```diff
@@ -1,68 +1,202 @@
...
+   * @description The text for the block, in the form of a text object. Minimum length for the `text` in this field is
+   * 1 and maximum length is 3000 characters. This field is not required if a valid array of `fields` objects is
+   * provided instead.
+   */
+  text?: PlainTextElement | MrkdwnElement;
+  /**
+   * @description Required if no `text` is provided. An array of text objects. Any text objects included with `fields`
+   * will be rendered in a compact format that allows for 2 columns of side-by-side text. Maximum number of items is 10.
+   * Maximum length for the text in each item is 2000 characters.
+   * {@link https://app.slack.com/block-kit-builder/T029V6468RL#%7B%22blocks%22:%5B%7B%22type%22:%22section%22,%22text%22:%7B%22text%22:%22A%20message%20*with%20some%20bold%20text*%20and%20_some%20italicized%20text_.%22,%22type%22:%22mrkdwn%22%7D,%22fields%22:%5B%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Priority*%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Type*%22%7D,%7B%22type%22:%22plain_text%22,%22text%22:%22High%22%7D,%7B%22type%22:%22plain_text%22,%22text%22:%22String%22%7D%5D%7D%5D%7D Click here for an example}.
```
Comment (seratch) [inline packages/types/src/block-kit/blocks.ts:194]: You can remove a team_id from the URL
```suggestion
   * {@link https://app.slack.com/block-kit-builder/#%7B%22blocks%22:%5B%7B%22type%22:%22section%22,%22text%22:%7B%22text%22:%22A%20message%20*with%20some%20bold%20text*%20and%20_some%20italicized%20text_.%22,%22type%22:%22mrkdwn%22%7D,%22fields%22:%5B%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Priority*%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Type*%22%7D,%7B%22type%22:%22plain_text%22,%22text%22:%22High%22%7D,%7B%22type%22:%22plain_text%22,%22text%22:%22String%22%7D%5D%7D%5D%7D Click here for an example}.
```

**Review by seratch — APPROVED** [review]

File: packages/types/src/block-kit/block-elements.ts [hunk 2]
```diff
@@ -1,65 +1,149 @@
...
    */
   initial_value?: string;
 }
 
+/**
+ * @description Displays an image as part of a larger block of content. Use this `image` block if you want a block with
+ * only an image in it.
+ * @see {@link https://api.slack.com/reference/block-kit/block-elements#image Image element reference}.
+ */
 export interface ImageElement {
```
Comment (srajiang) [inline packages/types/src/block-kit/block-elements.ts:135]: Could take a similar approach later of aliasing `ImageElement` -> `Image` as proposed for ConfirmationDialog down below with the motivation of lining up the terms between types and api.slack.com

File: packages/types/src/block-kit/block-elements.ts [hunk 3]
```diff
@@ -68,102 +152,362 @@ export interface ImageElement {
...
+   * @description Specifies the maximum number of items that can be selected in the menu. Minimum number is 1.
+   */
   max_selected_items?: number;
 }
 
 /*
  * End of select/multi-select menus
  */
 
+// TODO: the action_id inherited from Action type here IS apparently optional, unlike everywhere else!
```
Comment (srajiang) [inline packages/types/src/block-kit/block-elements.ts:499]: maybe a docs typo? 🤔

**Review by srajiang — APPROVED** [review]
LGTM!

File: packages/types/src/block-kit/block-elements.ts (hunk 2 above)
Comment (filmaj) [reply packages/types/src/block-kit/block-elements.ts:135]: I considered this as well but there's, confusingly, both an image block as well as an image block element ...

File: packages/types/src/block-kit/extensions.ts [hunk 4]
```diff
@@ -1,6 +1,8 @@
 // This file contains reusable extensions/mixins that other Block Kit elements will extend from.
 import { Confirm, PlainTextElement, DispatchActionConfig } from './composition-objects';
 
+// TODO: what about aliasing this to `Actionable` to line up the mixin terms in this file?
```
[the remark this answers is not in the document]
Comment (srajiang) [reply packages/types/src/block-kit/extensions.ts:4]: I like this!

**Pull request merged**